src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -129,7 +140,12 @@ internal Binder Binder
             SyntaxToken identifierToken,
             CSharpSyntaxNode context)
         {
-            return new OutLocalSymbol(containingSymbol, scopeBinder, enclosingBinderOpt, typeSyntax, identifierToken, context);
+            if (typeSyntax.IsVar)
+            {
```

Comment:
That is a reasonable design change that should be evaluated for all locals, not prematurely optimized for the one place it is least likely to have a positive impact on the compiler.